### PR TITLE
R2P-35:movers

### PR DESCRIPTION
**Pull request by ErnestoFolting**
Add 2 BaseMover and 2 inheritors movers. Replaced StatsController to PlayerBrain.

### CODE REVIEW

File: Platformer/Assets/Scripts/Movement/Controller/Movers/BaseMover.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using Movement.Data;
+using Movement.Enums;
+using UnityEngine;
```
Comment (voonya) [inline Platformer/Assets/Scripts/Movement/Controller/Movers/BaseMover.cs]: order of using

File: Platformer/Assets/Scripts/Movement/Controller/Movers/PositionMover.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using Movement.Controller.Movers;
+using Movement.Enums;
+using UnityEngine;
```
Comment (voonya) [inline Platformer/Assets/Scripts/Movement/Controller/Movers/PositionMover.cs]: order

File: Platformer/Assets/Scripts/Movement/Controller/Movers/PositionMover.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Movement.Controller
+{
+    public class PositionMover : BaseMover
+    {
+        private Vector2 _destination;
```
Comment (voonya) [inline Platformer/Assets/Scripts/Movement/Controller/Movers/PositionMover.cs:10]: maybe first should go public fields?

File: Platformer/Assets/Scripts/Movement/Controller/Movers/BaseMover.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace Movement.Controller.Movers
+{
+    public abstract class BaseMover
+    {
+        protected readonly Rigidbody2D Rigidbody;
+        public Direction Direction { get; private set; }
```
Comment (voonya) [inline Platformer/Assets/Scripts/Movement/Controller/Movers/BaseMover.cs:11]: maybe first public?

File: Platformer/Assets/Scripts/Movement/Controller/Movers/VelocityMover.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
+﻿using Movement.Enums;
+using UnityEngine;
+
+namespace Movement.Controller.Movers
+{
+    public class VelocityMover : BaseMover
+    {
+        private Vector2 _movement;
+        public override bool MoveActive => Rigidbody.velocity.x != 0;
```
Comment (voonya) [inline Platformer/Assets/Scripts/Movement/Controller/Movers/VelocityMover.cs:10]: maybe first should go public fields?

File: Platformer/Assets/Scripts/Movement/Enums/Direction.cs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Movement.Enums
+{
+    public enum Direction
+    {
+        Right = 0,
+        Left = 1,
+    }
+}
```
Comment (x-FIREFOUR-x) [inline Platformer/Assets/Scripts/Movement/Enums/Direction.cs]: empty line

File: Platformer/Assets/Scripts/Movement/Controller/Movers/BaseMover.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+        public abstract void Move(float horizontalMovement, bool isCanMove);
+
+        public void SetDirection(Direction newDirection)
+        {
+            if (Direction == newDirection) 
+                return;
+            
+            Rigidbody.transform.Rotate(0, 180, 0);
+            Direction = Direction == Direction.Right ? Direction.Left : Direction.Right;
```
Comment (x-FIREFOUR-x) [inline Platformer/Assets/Scripts/Movement/Controller/Movers/BaseMover.cs]: Maybe `Direction = newDirection` ?

**Review by voonya — APPROVED** [review]

**Review by x-FIREFOUR-x — APPROVED** [review]

**Pull request merged**